core/domain/html_cleaner.py

```diff
@@ -27,9 +27,18 @@
 
 import bleach
 import bs4
+from typing import Any, Dict, List
+from typing_extensions import Final, TypedDict
 
 
-def filter_a(tag, name, value):
+class ComponentsDict(TypedDict, total=False):
```

Comment:
Why is `total=False`? If it is to avoid the possible mypy error in function `get_rte_components`. Wouldn't it better if the component declaration in line 180 be done right after `customization_args` is formed?

Reply:
I guess then also we will get similar errors like this. 
### error: Argument 1 to "append" of "list" has incompatible type "Dict[str, Any]"; expected "ComponentsDict"  [arg-type]
I tried what you said but still getting this error. Probably because Mypy is assuming the type of `components` as Dict[str, Any]. So, should I keep that `Total`?

Reply:
I think `total` is unnecessary. We just need to refactor the code a bit.

```
 component: ComponentsDict = {
         'id': tag_name,
         'customization_args': customization_args
  }
  components.append(component)
```

Also the typing in line 181 `customization_args: Dict[str, Any] = {}` wont be required I guess if we type the component before appending it.

Reply:
thanks for the suggestion @aasiffaizal. I have updated accordingly, PTAL.